### PR TITLE
Added assemblies upload support

### LINKED ISSUE #81: Assemblies upload
## Describe your idea for a new feature
Be able to upload assemblies directly.

## Additional information
Probably build it off of the directory upload to start.

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## Description of changes
Added support for uploading assemblies. Use `-a` / `--assemblies` or the checkbox on the GUI

Added a `get_assemblies_files` method to `api_calls.py`, this is only used in test cases to confirm end to end upload.

## Related issue
Fixes #81 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: docs/developers/api.md [hunk 1]
```diff
@@ -54,6 +54,19 @@ project_id -- the id of the project the sample is on
 
 list of sequencefile dictionary for given sample_id
 
+#### get_assemblies_files(self, project_id, sample_name)
+API call to api/projects/project_id/sample_id/assemblies
+We fetch the assemblies files through the project id on this route
+
+**arguments:**
+
+sample_name -- the sample id to get from irida, relative to a project
```
Comment (deepsidhu85) [inline docs/developers/api.md:63]: If it is the sample id to get from irida shouldn't this read sample_id? If it is supposed to be sample_name add more to this line to make it clear

File: iridauploader/api/api_calls.py [hunk 2]
```diff
@@ -433,6 +431,52 @@ def get_sequence_files(self, project_id, sample_name):
...
         return result
 
+    def get_assemblies_files(self, project_id, sample_name):
+        """
+        API call to api/projects/project_id/sample_id/assemblies
+        We fetch the assemblies files through the project id on this route
+
+        arguments:
+
+            sample_name -- the sample id to get from irida, relative to a project
```
Comment (deepsidhu85) [inline iridauploader/api/api_calls.py:441]: Same as comment above about clarity

File: docs/developers/api.md (hunk 1 above)
Comment (JeffreyThiessen) [reply docs/developers/api.md]: Added some clarity here and in other places in the api that were a bit misleading,

It is actually the sample name that is used as the identifier, and not the sample id number.

File: iridauploader/api/api_calls.py (hunk 2 above)
Comment (JeffreyThiessen) [reply iridauploader/api/api_calls.py]: Done.

Comment (deepsidhu85) [reply iridauploader/api/api_calls.py]: Thanks!

**Pull request merged**